stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -218,12 +221,24 @@ public BStruct generateNext() {
         return bStruct;
     }
 
+    private void validateAndSetUnionRefRecordField(BStruct bStruct, int refRegIndex, int expectedTypeTag,
+            int actualTypeTag, BRefType value) {
+        validateAndSetRefRecordField(bStruct, refRegIndex, expectedTypeTag, actualTypeTag, value,
+                UNASSIGNABLE_UNIONTYPE_EXCEPTION);
+    }
+
+    private void validateAndSetNonUnionRefRecordField(BStruct bStruct, int refRegIndex, int expectedTypeTag,
+            int actualTypeTag, BRefType value) {
+        validateAndSetRefRecordField(bStruct, refRegIndex, expectedTypeTag, actualTypeTag, value,
+                MISMATCHING_FIELD_ASSIGNMENT);
+    }
```

Comment:
Do we need this extra level of indirection provided by the above two functions? Since those are only differ by constant string parameter , we can directly call validateAndSetRefRecordField where ever the above two functions are used.  IMO since no complex logics are used within the body, this indirection is increasing the complexity and reduce the readability of the code.